internal/inputs/http.go

```diff
@@ -246,50 +246,40 @@ func setRequestOptions(request *gorequest.SuperAgent, yml load.Config, api load.
 }
 
 // handleJSON Process JSON Payload
-func handleJSON(dataStore *[]interface{}, body []byte, resp *gorequest.Response, doLoop *bool, url *string, nextLink string) {
-	var f interface{}
-	err := json.Unmarshal(body, &f)
+func handleJSON(sample *[]interface{}, body []byte, resp *gorequest.Response, doLoop *bool, url *string, nextLink string, includeHeaders bool) {
+	var b interface{}
+	err := json.Unmarshal(body, &b)
+
 	if err != nil {
 		load.Logrus.WithError(err).Error("http: failed to unmarshal json")
-	} else {
-		switch f := f.(type) {
-		case []interface{}:
-			for _, sample := range f {
-
-				switch sample := sample.(type) {
-				case map[string]interface{}:
-					httpSample := sample
-					httpSample["api.StatusCode"] = (*resp).StatusCode
-					*dataStore = append(*dataStore, httpSample)
-					// load.StoreAppend(httpSample)
-				case string:
-					strSample := map[string]interface{}{
-						"output": sample,
-					}
-					// load.StoreAppend(strSample)
-					*dataStore = append(*dataStore, strSample)
-				default:
-					load.Logrus.Debugf("http: unsupported sample type: %T %v", sample, sample)
-				}
-			}
+		return
+	}
 
-		case map[string]interface{}:
-			theSample := f
-			theSample["api.StatusCode"] = (*resp).StatusCode
-			// load.StoreAppend(theSample)
-			*dataStore = append(*dataStore, theSample)
+	var cb responseBody
 
-			if theSample["error"] != nil && fmt.Sprintf("%v", theSample["error"]) != "false" {
-				load.Logrus.Debugf("http: request failed %v", theSample["error"])
-			}
+	switch t := b.(type) {
+	case []interface{}:
+		cb = newObjectBody(t)
```

Comment:
shouldn't this be the other way around? In this case the type `t` is a array on "objects" but you are using 
`newObjectBody` instead of `newArrayBody()`?

Reply:
looks like the functions just needs to be renamed?

Reply:
Yes, indeed. Thanks